central/networkgraph/entity/gatherer/online_gatherer.go

```diff
@@ -29,23 +29,30 @@ var (
 		sac.AllowFixedScopes(
 			sac.AccessModeScopeKeys(storage.Access_READ_WRITE_ACCESS),
 			sac.ResourceScopeKeys(resources.NetworkGraph)))
+
+	blobAccessCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.Administration)))
 )
 
 type defaultExtSrcsGathererImpl struct {
 	networkEntityDS entityDataStore.EntityDataStore
 	stopSig         concurrency.Signal
+	blobStore       blobstore.Datastore
 }
 
 // newDefaultExtNetworksGatherer returns an instance of NetworkGraphDefaultExtSrcsGatherer that reaches out internet to fetch the data.
 func newDefaultExtNetworksGatherer(networkEntityDS entityDataStore.EntityDataStore) NetworkGraphDefaultExtSrcsGatherer {
 	return &defaultExtSrcsGathererImpl{
 		networkEntityDS: networkEntityDS,
+		blobStore:       blobstore.Singleton(),
```

Comment:
I don't think this is the right place to put the singleton and should be wired into the gatherer singleton instead